File: Kaskus/GridViewPaging/GridViewPagingCs/Pages/DataSourceResult.cs
namespace Dede.App.GridViewPaging.Pages
{
    using System;
    using System.Collections;

    public abstract class DataSourceResult
    {
        protected DataSourceResult()
        {
            this.TotalRecord = 0;
        }

        public Int32 TotalRecord { get; set; }

        public IList GetList()
        {
            return this.MyList();
        }

        protected abstract IList MyList();
    }
}

File: Kaskus/GridViewPaging/GridViewPagingVb/Pages/DataSourceResult.vb
Imports System
Imports System.Collections

Namespace Pages

    Public MustInherit Class DataSourceResult
        Protected Sub New()
            TotalRecord = 0
        End Sub

        Public Property TotalRecord() As Int32

        Public Function GetList() As IList
            Return MyList()
        End Function

        Protected MustOverride Function MyList() As IList
    End Class
End Namespace